libqtile/backend/base.py

```diff
@@ -253,6 +256,65 @@ def info(self) -> dict[str, Any]:
         """
         return {}
 
+    @property
+    def kept_above(self) -> bool:
+        return False
+
+    @kept_above.setter
+    def kept_above(self, value: bool) -> None:
+        pass
+
+    @expose_command()
+    def keep_above(self) -> None:
+        """Keep this window above all others"""
+
+    @expose_command()
+    def keep_below(self) -> None:
+        """Keep this window below all others"""
+
+    @expose_command()
+    def move_up(self, force: bool = False) -> None:
+        """
+        Move this window above the next window along the z axis.
+
+        Will not raise a window where "keep_below" is True unless
+        force is set to True.
+        """
+
+    @expose_command()
+    def move_down(self, force: bool = False) -> None:
+        """
+        Move this window below the previous window along the z axis.
+
+        Will not lower a window where "keep_above" is True unless
+        force is set to True.
+        """
+
+    @expose_command()
+    def move_to_top(self) -> None:
+        """
+        Move this window above all windows in the current layer
+        e.g. if you have 3 windows all with "keep_above" set, calling
+        this method will move the window to the top of those three windows.
+
+        Calling this on a "normal" window will not raise it above a "kept_above"
+        window.
+        """
+
+    @expose_command()
+    def move_to_bottom(self) -> None:
+        """
+        Move this window below all windows in the current layer
+        e.g. if you have 3 windows all with "keep_above" set, calling
+        this method will move the window to the bottom of those three windows.
+
+        Calling this on a "normal" window will not raise it below a "kept_below"
+        window.
+        """
+
+    def change_layer(self, up: bool = True, top_bottom: bool = False, kwargs=None) -> None:
+        """Trigger repositioning of window in layer stack."""
```

Comment:
This appears to be an implementation detail of the X backend. Could we move it to that backend instead of here?

Reply:
Ok. We also have things like `check_stacking` in `manager.py` and `update_client_list/stack` in `base/backend.py`. Does that all need to get moved to the x11 core?

`change_layer` is also called from `manage` in `manager.py` so, if we move it, we need to rethink how it's triggered when windows are managed.